src/pipeline.ml

```diff
@@ -57,223 +57,250 @@ let include_git = { Cluster_api.Docker.Spec.defaults with include_git = true }
+      ]
+    in
+    pipelines, additional_build_args
 
-  let additional_build_args =
-    let+ opam_repository_commit =
-      head_of { Github.Repo_id.owner = "ocaml"; name = "opam-repository" } @@ `Ref "refs/heads/master"
-    and+ platform_blog_commit =
-      head_of { Github.Repo_id.owner = "ocaml"; name = "platform-blog" } @@ `Ref "refs/heads/master"
-    and+ opam_commit =
-      head_of { Github.Repo_id.owner = "ocaml"; name = "opam" } @@ `Ref "refs/heads/master" in
-    ["OPAM_REPO_GIT_SHA=" ^ Current_git.Commit_id.hash opam_repository_commit;
-     "BLOG_GIT_SHA=" ^ Current_git.Commit_id.hash platform_blog_commit;
-     "OPAM_GIT_SHA=" ^ Current_git.Commit_id.hash opam_commit]
-  in
+  let extras () =
+    let tarsnap =
+      let monthly = Current_cache.Schedule.v ~valid_for:(Duration.of_day 30) () in
+      Current_ssh.run ~schedule:monthly "watch.ocaml.org" ~key:"tarsnap" (Current.return ["./tarsnap-backup.sh"])
+    in
+    let peertube =
+      let weekly = Current_cache.Schedule.v ~valid_for:(Duration.of_day 7) () in
+      let image = Cluster.Watch_docker.pull ~schedule:weekly "chocobozzz/peertube:production-bookworm" in
+      Cluster.Watch_docker.service ~name:"infra_peertube" ~image ()
+    in
+    [tarsnap; peertube]
 
-  let opam_repository_pipeline = filter_list filter [
-    ocaml_opam, "opam2web", [
-      docker_registry (Duration.of_min 360) "Dockerfile"
-      ["live", "opam.ocaml.org:live", [`Ocamlorg_opam4 "infra_opam_live"; `Ocamlorg_opam5 "infra_opam_live"];
-       "live-staging", "opam.ocaml.org:staging", [`Ocamlorg_opam4 "infra_opam_staging"; `Ocamlorg_opam5 "infra_opam_staging"]]
-    ];
-  ]
-  in
+  let v ?app ?notify:channel ?filter ~sched ~staging_auth () =
+    (* [web_ui collapse_value] is a URL back to the deployment service, for links
+      in status messages. *)
+    let web_ui repo = Uri.with_query' base_url ["repo", repo] in
+    let build ?additional_build_args (org, name, builds) =
+      Cluster_build.repo ?channel ?additional_build_args ~web_ui ~org ~name builds
+    in
+    let build_for_registry ?additional_build_args (org, name, builds) =
+      Build_registry.repo ?channel ?additional_build_args ~web_ui ~org ~name builds
+    in
+    let pipelines = filter_list filter @@ services ?app ~sched ~staging_auth () in
 
-  let monthly = Current_cache.Schedule.v ~valid_for:(Duration.of_day 30) () in
-  let tarsnap = Current_ssh.run ~schedule:monthly "watch.ocaml.org" ~key:"tarsnap" (Current.return ["./tarsnap-backup.sh"])
-  in
-  let peertube =
-    let weekly = Current_cache.Schedule.v ~valid_for:(Duration.of_day 7) () in
-    let image = Cluster.Watch_docker.pull ~schedule:weekly "chocobozzz/peertube:production-bookworm" in
-    Cluster.Watch_docker.service ~name:"infra_peertube" ~image ()
+    let opam_repository_pipelines, additional_build_args =
+      opam_repository ?app ()
+      |> (fun (pipelines, args) -> filter_list filter @@ pipelines, args)
```

Comment:
iiuc, this would be clearer as something like:

```suggestion
    let opam_repository_pipelines, additional_build_args = opam_repository ?app () in
    let filtered_opam_repository_pipelines = filter_list filter opam_repository_pipelines in
```

Reply:
Good suggestion